### DISCUSSION TITLE
How can I program a button?

### DISCUSSION THREAD

**Discussion by shirakuru** [TomSchimansky/CustomTkinter#1636] [Q&A]
How can I program a button so that when it is clicked, a new window opens?I tried many ways, but nothing happened when I pressed the button. Please help.
In this case, I need to program this button: 
self.home_frame_button_2 = customtkinter.CTkButton(self.home_frame, text="AUDCAD", image=self.image_icon_image,
                                                           compound="right", height=50, width=300,
                                                           command=self.command)
        self.home_frame_button_2.grid(row=2, column=0, padx=20, pady=10)

**Comment by alfmelchor** [comment]
You can always use CTkTopLevel for additional windows which are part of your app. However, if you're referring to creating a completely new master, you need to create the function you're referring to in your `self.command`. 

```
def command():
    newroot = customtkinter.CTk()
    # Implement all your code for the new root here.
    newroot.mainloop()
```
Keep in mind if you destroy one window, the new one will destroy. You will need to change the way your code structure is laid out.

**Reply by TomSchimansky** [reply]
You can not run multiple instances of customtkinter.CTk() in parallel, always use CTkToplevel. This can cause various problems as reported by some users in the issues.
